subsys/fs/fuse_fs_access.c

```diff
@@ -178,7 +179,9 @@ static int fuse_fs_access_readdir(const char *path, void *buf,
 		 */
 		char mount_path[PATH_MAX];
 
-		sprintf(mount_path, "%s/", path);
+		/* We need space for '/' and '\0' */
+		sprintf(mount_path, "%.*s/", MIN(strlen(path),
+					(PATH_MAX - 2)), path);
```

Comment:
I think it would be more proper to check first whether the additional '/' fits with string to the mount point and use `strcpy`, or similar rather than heavy snprintf, and fill in missing '/'.
Above is also dangerous because you can get path that is still existing path, but not a path you want to access.
For example you can cut something like "/bi/ba/bu" to "/bi/ba//" which is still valid path.

Reply:
yep ... thought about it but decided to just truncate the contents and send to the upper level to keep it as closes as the original implementation ...

But it is probably better just return an error here and avoid propagate inaccurate data.

Reply:
updated